test/distributed/_composable/fully_shard/test_fully_shard_mixed_precision.py

```diff
@@ -31,59 +31,160 @@
 class TestMixedPrecision(FSDPTest):
     """Tests ``fully_shard`` with mixed precision."""
 
+    EXPECTED_CAST_DTYPES = {
+        # (cast_root_forward_inputs_submodule, cast_forward_inputs_submodule, use_root_no_params)
```

Comment:
IIUC, this comment is for the keys. Should we also explain here what the values represent? Alternatively, we could put it into a `NamedTuple`, but that is not strictly required.

Reply:
Good suggestion! Using `NamedTuple` s is clearer for defining the subtest configurations (as well as in accessing them where appropriate). Also, with `NamedTuple` s, we can minimize the config a bit by only specifying non-default values, in this case, the variables that should be full precision.